### ISSUE TITLE
Numpy Array Type Error in Scatter() function (related to Issue #84)

### ISSUE THREAD

**Issue by mmguar** [piccolomo/plotext#85]
```
x = np.array([0,1,2])
y = np.array([1,2,3])
plt.scatter(x,y)
plt.show()
```
Returns the following error:
`Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/matteo/anaconda3/lib/python3.9/site-packages/plotext/_figure.py", line 286, in show
    self.build()
  File "/home/matteo/anaconda3/lib/python3.9/site-packages/plotext/_figure.py", line 274, in build
    self.subplot.update_matrix()
  File "/home/matteo/anaconda3/lib/python3.9/site-packages/plotext/_subplot.py", line 330, in update_matrix
    y0 = [max(0, el) for el in y0]
  File "/home/matteo/anaconda3/lib/python3.9/site-packages/plotext/_subplot.py", line 330, in <listcomp>
    y0 = [max(0, el) for el in y0]
TypeError: '>' not supported between instances of 'NoneType' and 'int'
`
but:
```
x = [0,1,2]
y = [1,2,3]
plt.scatter(x,y)
plt.show()
```
works.
Interestingly:`
y = np.array([XXXXXX])`
plt.scatter(y)
plt.show()`
does not give an error but the plot is not always correct (in terms of the order of the points).

**Comment by mmguar** [comment]
I understand this is similar to issue #84 , but some subtleties are a little different, so it may warrant double testing the solution against this use case too.
Feel free to close if the proposed solution works.

**Comment by piccolomo** [comment]
Hi @mmguar, thanks a lot for the post. Thanks to the similar input in Issue 84. I have implemented the following change to [this function](https://github.com/piccolomo/plotext/blob/0e3f8b18046f7a68fda2bd40be8a75c0cf1bc0bb/plotext/_utility/data.py#L11)

```python
def set_data(x = None, y = None): # it return properly formatted x and y data lists
   if x is None and y is None :
       x, y = [], []
   elif x is not None and y is None:
       y = x
       x = range(1, len(y) + 1)
   x, y = remove_non_numerical(x, y)
   lx, ly = len(x), len(y)
   if lx != ly:
       l = min(lx, ly)
       x = x[ : l]
       y = y[ : l]
   return [x, y]

non_numerical = lambda el: el == None or math.isnan(el)

def remove_non_numerical(x, y):
   l = len(x)
   p = [i for i in range(l) if not (non_numerical(x[i]) or non_numerical(y[i]))]
   xn = [float(x[i]) for i in p]
   yn = [float(y[i]) for i in p]
   return xn, yn
```

which is not yet published, but seem to work fine with `numpy` arrays.

**Comment by piccolomo** [comment]
Hello @mmguar ,

Issue solved in [version 4.3](https://github.com/piccolomo/plotext) available only on GitHub for now, using:

```
pip install git+https://github.com/piccolomo/plotext
```

Any feedback or double test is appreciated.

Thanks,
Savino

**Comment by mmguar** [comment]
It looks good after a quick test. Thanks for the quick fix!
